### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by faraz-g**
Hi there, my student number is 45306516. This pull request contains the completed GAN model for the OASIS dataset. The jupyter notebook containing my model is inside the "OASIS DCGAN.ipynb" file. The model, data processing, and driver script have been separated into the "model" folder for ease of use. 

Please refer to the README file for a breakdown of the model, and the usage guide.

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Commit messages should be more specific - what exactly was changed?
- Code valid for Hard Difficulty
- Good results 
- The entire note book is uncommented

Other Feedback
- Please group the `.png` files and put them in a sub-directory
- I'd put the model code in an external file and import them

**Comment by rexsw** [comment]
Code looks good for hard difficulty

For next time;

Tensorflow has functions for image opening and more so there’s no need to also use imageio in your project.

File: recognition/s4530651-OASIS-DCGAN/model/data.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import glob
```
Comment (shakes76) [inline recognition/s4530651-OASIS-DCGAN/model/data.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. remove unnecessary epoch PNG files.**
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Pull request closed without merging**